src/Compilers/CSharp/Portable/FlowAnalysis/ReadWriteWalker.cs

```diff
@@ -220,7 +220,28 @@ public override void VisitForEachIterationVariable(BoundForEachStatement node)
 
         public override BoundNode VisitRangeVariable(BoundRangeVariable node)
         {
-            NoteRead(node.RangeVariableSymbol);
+            // Compute the "underlying symbol" for a read of the range variable
+            Symbol underlyingSymbol = GetUnderlyingVariable(node.Value);
+            NoteRead(node.RangeVariableSymbol, underlyingSymbol);
+            return null;
+        }
+
+        private Symbol GetUnderlyingVariable(BoundNode underlying)
```

Comment:
It looks like this method can return only ParameterSymbol or null. Consider changing return type to ParameterSymbol. Also, consider adding a comment about the purpose of the method and possibly renaming it to GetRangeVariableUnderlyingParameter because it never returns a variable. Consider renaming new parameter in other APIs to rangeVariableUnderlyingParameter and using ParameterSymbol as its type.